src/components/TooltipNew/UserGuideTooltip.js

```diff
@@ -0,0 +1,173 @@
+        return {
+          shouldSlide: false,
+          lastElement: props.element
+        };
+      }
+
+      if (props.element !== state.lastElement) {
+        // the element has changed - slide to the next one
+        return {
+          shouldSlide: true,
+          lastElement: props.element
+        };
+      }
+
+      // window was scrolled or any other change has happened - don't slide
+      return {
+        shouldSlide: false
+      };
+    }
+
+    // reset the element when the tooltip hides
+    return {
+      lastElement: null
+    };
+  }
+
+  constructor(props) {
+    super(props);
+
+    // we don't use the following state params but we still need to re-render when the window changes
+    /* eslint-disable react/no-unused-state */
+    this.state = {
+      windowWidth: 0,
+      windowHeight: 0,
+      windowScrollX: 0,
+      windowScrollY: 0
+    };
+    /* eslint-enable react/no-unused-state */
+
+    this.handleDocumentChange = throttle(
+      16,
+      this.handleDocumentChange.bind(this)
+    );
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleDocumentChange);
+    document.addEventListener('scroll', this.handleDocumentChange);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleDocumentChange);
+    document.removeEventListener('scroll', this.handleDocumentChange);
+  }
+
+  handleDocumentChange() {
+    const windowWidth = window.innerWidth;
+    const windowHeight = window.innerHeight;
+    const windowScrollX = document.documentElement.scrollLeft;
+    const windowScrollY = document.documentElement.scrollTop;
+
+    /* eslint-disable react/no-unused-state */
+    this.setState({
+      windowWidth,
+      windowHeight,
+      windowScrollX,
+      windowScrollY
+    });
+    /* eslint-enable react/no-unused-state */
+  }
+
+  render() {
+    const {
+      children,
+      className,
+      placement,
+      zIndex,
+      element,
+      isVisible,
+      slide,
+      theme,
+      containerName
+    } = this.props;
+    const referenceElement = memoizedReference(element, 8);
+    const rect = referenceElement.getBoundingClientRect();
+    const shouldSlide = slide && this.state.shouldSlide;
+
+    return (
+      <ModalPortal parentElementName={containerName}>
+        <SpotlightOverlay
+          gap={rect}
+          isVisible={isVisible}
+          slide={shouldSlide}
+        />
+        <PopperTooltip
+          theme={theme || 'invert'}
+          className={cx({
+            [styles[baseClass]]: true,
+            [styles[`${baseClass}--slide`]]: shouldSlide,
+            [className]: className
+          })}
+          placement={placement}
+          triggerActionType="managed"
+          isVisible={isVisible}
+          referenceElement={referenceElement}
+          zIndex={zIndex}
+          modifiers={{
+            offset: {
+              offset: '0, 20'
```

Comment:
a new `modifiers` object will be created on each render,
please declare modifiers as class property or constant above class

Reply:
I created a separate const